pkg/syncer/metadatasyncer.go

```diff
@@ -124,7 +125,11 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
 		return err
 	}
 	// Initialize the k8s orchestrator interface
-	metadataSyncer.coCommonInterface, err = commonco.GetContainerOrchestratorInterface(ctx, common.Kubernetes, metadataSyncer.configInfo.Cfg.FeatureStatesConfig)
+	k8sInitParams := k8sorchestrator.K8sInitParams{
```

Comment:
`k8sInitParams`  can be created according to the clusterFlavor. For example, there is no need to initialize `SupervisorFeatureStatesConfigInfo` for vanilla, etc.

Reply:
Changed it